sympy/integrals/intpoly.py

```diff
@@ -294,21 +321,18 @@ def main_integrate(expr, facets, hp_params, max_degree=None):
                                 (b / norm(a)) / (dim_length + degree)
         return result
     else:
-        polynomials = decompose(expr)
-        for deg in polynomials:
-            poly_contribute = S.Zero
-            facet_count = 0
-            for hp in hp_params:
-                value_over_boundary = integration_reduction(facets,
-                                                            facet_count,
-                                                            hp[0], hp[1],
-                                                            polynomials[deg],
-                                                            dims, deg)
-                poly_contribute += value_over_boundary * (hp[1] / norm(hp[0]))
-                facet_count += 1
-            poly_contribute /= (dim_length + deg)
-            integral_value += poly_contribute
-    return integral_value
+        if not isinstance(expr, list):
+            polynomials = decompose(expr)
+            answer = _polynomial_integrate(polynomials, facets, hp_params)
+            return answer
+        else:
+            result = {}
+            for exp in expr:
+                polynomials = decompose(exp)
+                answer = _polynomial_integrate(polynomials, facets, hp_params)
+                result.update({exp : answer})
+
+            return result
```

Comment:
```suggestion
            return {exp: _polynomial_integrate(decompose(exp), facets, hp_params) for exp in expr}
```
is a bit more compact/pythonic, but no big deal.

Reply:
And maybe a bit too long for a single line...

Reply:
Thanks for the suggestions @oscargus  ,I tried to mimic the earlier format/style of code .For the first time I've committed suggestions directly from here (without changing my branch) , hope that it works fine !